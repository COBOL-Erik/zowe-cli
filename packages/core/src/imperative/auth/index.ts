/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

export * from "./builders/AuthCommandBuilder";
export * from "./builders/AuthLoginCommandBuilder";
export * from "./builders/AuthLogoutCommandBuilder";
export * from "./builders/CompleteAuthGroupBuilder";

export * from "./doc/IAuthHandlerApi";

export * from "./handlers/AbstractAuthHandler";
export * from "./handlers/BaseAuthHandler";
