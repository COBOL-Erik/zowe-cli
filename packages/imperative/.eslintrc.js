module.exports = {
    ...require("../../.eslintrc.js"),
    "parserOptions": {
        "ecmaVersion": 12,
        "sourceType": "module",
        "project": [
            "./tsconfig.json",
            "./__tests__/tsconfig.json",
            "./tsconfig-tests.json"
        ]
    }
}