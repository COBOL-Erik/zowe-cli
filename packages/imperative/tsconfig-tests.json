{
  "compilerOptions": {
    "types": [
      "node",
      "jest"
    ],
    "lib": [
      "esnext"
    ],
    "experimentalDecorators": true,
    "target": "es2015",
    "module": "commonjs",
    "noEmit": true,
    "declaration": true,
    "moduleResolution": "node",
    "noImplicitAny": true,
    "outDir": "./lib",
    "preserveConstEnums": true,
    "removeComments": false,
    "pretty": true,
    "sourceMap": true,
    "newLine": "lf"
  },
  "typeRoots": [
    "./node_modules/@types"
  ],
  "include": [
    "**/__tests__/*",
    "**/__tests__/**/*",
    "**/__mocks__/*"
  ],
  "exclude": [
    "lib",
    "node_modules"
  ],
  "files": [
    "../../__types__/wontache.d.ts"
  ],
  "allowJs": true
}
