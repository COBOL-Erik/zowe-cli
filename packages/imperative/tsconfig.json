{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "./lib",
    "stripInternal": true
  },
  "include": [
    "./src"
  ]
}