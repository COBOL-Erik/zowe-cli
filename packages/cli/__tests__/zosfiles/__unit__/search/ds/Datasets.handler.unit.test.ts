/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/


import { Search } from "@zowe/zos-files-for-zowe-sdk";
import { UNIT_TEST_ZOSMF_PROF_OPTS } from "../../../../../../../__tests__/__src__/mocks/ZosmfProfileMock";
import { TaskStage } from "@zowe/imperative";

describe("Search Datasets handler", () => {
    describe("process method", () => {
        it("should search a data set if requested", async () => {
            // Require the handler and create a new instance
            const handlerReq = require("../../../../../src/zosfiles/search/ds/DataSets.handler");
            const handler = new handlerReq.default();
            const dataSetName = "TEST*";
            const searchString = "test";

            // Vars populated by the mocked function
            let apiMessage = "";
            let jsonObj;
            let logMessage = "";
            let fakeSession = null;

            // Mock the submit JCL function
            Search.dataSets = jest.fn(async (session) => {
                fakeSession = session;
                return {
                    success: true,
                    commandResponse: "Found \"test\" in 2 data sets and PDS members",
                    apiResponse: [
                        {
                            dsname: "TEST1.DS",
                            memname: "TESTMEM",
                            matchList: [
                                {
                                    line: 1,
                                    column: 1,
                                    contents: "TEST CONTENTS"
                                }
                            ]
                        },
                        {
                            dsname: "TEST2.DS",
                            memname: undefined,
                            matchList: [
                                {
                                    line: 1,
                                    column: 1,
                                    contents: "TEST CONTENTS"
                                }
                            ]
                        }
                    ]
                };
            });

            // Invoke the handler with a full set of mocked arguments and response functions
            await handler.process({
                arguments: {
                    $0: "fake",
                    _: ["fake"],
                    dataSetName,
                    searchString,
                    ...UNIT_TEST_ZOSMF_PROF_OPTS
                },
                response: {
                    data: {
                        setMessage: jest.fn((setMsgArgs) => {
                            apiMessage = setMsgArgs;
                        }),
                        setObj: jest.fn((setObjArgs) => {
                            jsonObj = setObjArgs;
                        })
                    },
                    console: {
                        log: jest.fn((logArgs) => {
                            logMessage += "\n" + logArgs;
                        })
                    },
                    progress: {
                        startBar: jest.fn((parms) => {
                            // do nothing
                        }),
                        endBar: jest.fn(() => {
                            // do nothing
                        })
                    }
                }
            } as any);

            expect(Search.dataSets).toHaveBeenCalledTimes(1);
            expect(Search.dataSets).toHaveBeenCalledWith(fakeSession, {
                dsn: dataSetName,
                searchString,
                caseSensitive: undefined,
                mainframeSearch: undefined,
                maxConcurrentRequests: undefined,
                timeout: undefined,
                progressTask: {
                    percentComplete: 0,
                    statusMessage: "Starting search...",
                    stageName: TaskStage.NOT_STARTED
                }
            });
            expect(jsonObj).toMatchSnapshot();
            expect(apiMessage).toMatchSnapshot();
            expect(logMessage).toMatchSnapshot();
        });
    });
});
