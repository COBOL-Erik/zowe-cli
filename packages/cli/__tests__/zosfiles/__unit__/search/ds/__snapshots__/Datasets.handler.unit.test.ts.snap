// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Datasets handler process method should search a data set if requested 1`] = `
Object {
  "apiResponse": Array [
    Object {
      "dsname": "TEST1.DS",
      "matchList": Array [
        Object {
          "column": 1,
          "contents": "TEST CONTENTS",
          "line": 1,
        },
      ],
      "memname": "TESTMEM",
    },
    Object {
      "dsname": "TEST2.DS",
      "matchList": Array [
        Object {
          "column": 1,
          "contents": "TEST CONTENTS",
          "line": 1,
        },
      ],
      "memname": undefined,
    },
  ],
  "commandResponse": "Found \\"test\\" in 2 data sets and PDS members",
  "success": true,
}
`;

exports[`Search Datasets handler process method should search a data set if requested 2`] = `""`;

exports[`Search Datasets handler process method should search a data set if requested 3`] = `
"
Found \\"test\\" in 2 data sets and PDS members"
`;
